### PR TITLE
Release - Publish release and delete branch

### PR DESCRIPTION
**Pull request by araratthehero**
## Description
Publish the draft release for a specific tag, create the tag and delete the branch.

## Checklist 
- [x] PR is labelled 
- [x] Changes are tested manually

COAND-1015

### CODE REVIEW

File: .github/workflows/finalize_release.yml [hunk 1]
```diff
@@ -0,0 +1,63 @@
...
+      - uses: actions/checkout@v4
+
+      # Publish Github release and the tag
+      - name: Publish Github release
+        uses: ncipollo/release-action@v1
+        env:
+          VERSION_NAME: ${{ needs.generate_version_name.outputs.version_name }}
+        with:
+          tag: ${{ env.VERSION_NAME }}
+          allowUpdates: true
```
Comment (OscarSpruit) [inline .github/workflows/finalize_release.yml:43]: What does this actually mean? Would publishing the same version name override the previous release?

Comment (araratthehero) [reply .github/workflows/finalize_release.yml:43]: Indeed. If `allowUpdates` is true, then the release gets updated if it already exists, based on `tag`. I also set the `updateOnlyUnreleased` flag to true, which will only update a release if it is not published yet.

Comment (OscarSpruit) [reply .github/workflows/finalize_release.yml:43]: Ahh I see, that makes sense

File: .github/workflows/finalize_release.yml [hunk 2]
```diff
@@ -0,0 +1,65 @@
...
+
+  remove_release_branch:
+    runs-on: ubuntu-latest
+    needs: publish_draft_release
+
+    steps:
+      - uses: actions/checkout@v4
+
+      - name: Remove release branch
+        uses: dawidd6/action-delete-branch@v3
```
Comment (OscarSpruit) [inline .github/workflows/finalize_release.yml:59]: Not sure what does action all does, but wouldn't it be easier to use plain git?

Comment (araratthehero) [reply .github/workflows/finalize_release.yml:59]: This action allows deleting multiple branches and has some filtering options, but in our case is indeed not necessary.

**Review by OscarSpruit — APPROVED** [review]

**Pull request merged**